JoeDodgson/Fitness-buddy
# Joe/feat/pug setup

# Summary of changes

* Moved HTML files into public/html-to-be-deleted. Once we are happy that pages display as required we can delete all HTML files and amend the pug files directly
* Created a new folder for pug files called 'views'
* Created pug files for each page
* Created pug files for common elements across pages: 
  - head (for common elements in the head tag)
  - navbar
  - footer
* Created placeholder js files in the public/js folder
* Changes to server.js file:
  - Required in path node package
  - Set Pug as the default templating engine
* Changes to html-routes.js
  - Added html routes for all pages we have created
  - Set pug renderer for each route
  - Created dummy data to test the rendering of non-static data on html paths:
       - results
       - details
       - favourite-exe
  - Changed login and signup pages to redirect to favourite-exe page if user is logged in

# Testing
* Read my code changes
* Connect to JAWSDB from local workbench
* Test deployment
* Ensure that all pages render ok. Some are far from perfect but we can make changes to this later. We don't have links between pages so use the html paths to check:
  - /login (note: this will redirect to /favourite-exe if you are logged in)
  - /signup (note: this will redirect to /favourite-exe if you are logged in)
  - /search
  - /results
  - /:id  (test using any number in place of 'id'. Note: it will always display the same exercise details but with an updated data-id attribute)
  - /favourite-exe
